### DISCUSSION TITLE
Transfer between accounts with different currencies

### DISCUSSION THREAD

**Discussion by fsobolev** [NickvisionApps/Denaro#130] [Ideas]
Implementing #93 will be great, but it will open one new problem: what if the user wants to make transfer between accounts with different currencies? I suggest this solution.
After transfer is started, the app checks if accounts use the same currency. If not, a new dialog is opened, allowing the user to set conversion rate. It will also be useful to have "Set automatically" button, so the app will search somewhere for the rate and fill entries. It should not be used automatically tho, only on click, because currencies can be something not from real world. It will also be useful if the app would remember last used rate for two given currencies.
An example:
Let's say I have an account to track my 💠 primogems in Genshin Impact. There is daily transaction for daily quests, transaction for abyss, for events etc. Then I decided to donate some real money into the game: I open my wallet account, start a transfer. The app sees that there are two different currencies: ₽→💠. I can't use automatic conversion because primogems is not a real currency. I fill the rate manually: 10:16 (just as example, I don't remember the real cost). And the app remembers that, so the next time it sees ₽→💠 transfer it automatically fills the dialog entries with 10 and 16 respectively. And so after transfer is complete, there's expense transaction for 1000₽ in my wallet and income transaction for 1600💠 in my Genshin account.

**Comment by zothma** [comment]
I like the idea, but I don't think adding automatic conversion rate is the good idea, because:

1. as you stated, some currencies are fictive like the primogems (or the incredible dabloons),
2. conversion rates change over time. Transferring money between accounts does not have a transfer date (which I think would have been great though if I did the transfer one year ago for instance, but that's another story) so the current conversion rate might not be the one from the actual moment of the transaction and money has no way to know it

I think that if you transfer money between different currencies you're already aware of the conversion rate and can manually set it.

What do you think about that?

**Reply by fsobolev** [reply]
I had a feeling that automatic conversion can be useful somehow, but the more I think of it, the easier it is to come up with examples when it's not helpful. So I guess you are right, it's kinda useless.

**Comment by nlogozzo** [comment]
Moved to #144
